### DISCUSSION TITLE
Branch component doesn't update itself for projects / folders opened in Windows paths under WSL.

### DISCUSSION THREAD

**Discussion by nishnat-rishi** [nvim-lualine/lualine.nvim#1310] [Q&A]
But it DOES update for folders opened anywhere other than the /mnt/* paths (Native Ubuntu paths). Is this a problem that others are facing? I google a bit but couldn't find someone encountering this issue or a solution.

I am using Ubuntu 24.04 inside WSL2 (Windows 11).

I was trying to dig into the issue, seems like a vim.uv.new_fs_event issue, because when I changed the branch component to use vim.uv.new_fs_poll(), the issue went away. I thought this shouldn't happen for WSL, but it seems like it happens regardless.

Assuming this is a real issue, it might be solved by changing the following line in `~.local/share/nvim/lazy/lualine.nvim/lua/lualine/components/git_branch.lua` to use more robust checking of the underlying OS, or by digging new_fs_event or WSL code:

```lua
-- Windows doesn't like file watch for some reason.
-- local file_changed = sep ~= '\\' and vim.loop.new_fs_event() or vim.loop.new_fs_poll() -- old line
local file_changed = vim.loop.new_fs_poll() -- I'm just using this for now, not a real solution
```

I'll try digging into this further, but just wanted some sanity checks on whether I'm just the only guy facing this issue.

Note: The reason I'm having to use /mnt/* in WSL is to open projects started before I started using WSL. A workaround is to just copy everything within Ubuntu's FS, but we're just dodging the issue at that point.

**Comment by dmbhatti** [comment]
Facing this as well
Branch updating immediately on repos cloned outside /mnt paths
But not on /mnt/<drive> paths
